backend/validators/grafana_service.py

```
@@ -241,6 +305,198 @@ def sync_network(cls, network):
             'shame': shame_count,
         }
 
+    @classmethod
+    def _record_history(cls, samples, now):
```

Ruff (0.15.20)

[warning] Missing return type annotation for classmethod `_record_history`

Add return type annotation: `None`

(ANN206)